torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -709,6 +709,13 @@ void AliasDb::makePointerTo(const Value* from, const Value* to) {
     return;
   }
 
+  // Special case: if `from` is an optional, `to` could be a None. Don't
```

Comment:
Could we handle this more generally by returning if `!shouldAnnotate(to)`

Reply:
Normally, having `from` be mutable and `to` be immutable would be a bug, so I prefer to throw in that case. But I think we should make that explicit with an assertion (which would have made this bug a lot easier to track down in the first place).